src/holo-crypto-wasm/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+mod utils;
+mod arg_checks;
+
+use wasm_bindgen::prelude::*;
+
+use holo_config_core::{admin_keypair_from, keypair_from, Keypair, PublicKey};
+
+// When the `wee_alloc` feature is enabled, use `wee_alloc` as the global
+// allocator.
+#[cfg(feature = "wee_alloc")]
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[derive(Debug)]
+enum HoloCryptoError {
+    Custom(String),
+}
+
+impl From<HoloCryptoError> for JsValue {
+    fn from(e: HoloCryptoError) -> Self {
+        JsValue::from_str(&format!("{:?}", e).to_string())
+    }
+}
+
+#[wasm_bindgen]
+pub struct AdminKeyPair(Keypair);
+
+#[wasm_bindgen]
+impl AdminKeyPair {
+    #[wasm_bindgen(constructor)]
+    pub fn new(email: &str, pass: &str, hc_pub_key: &str) -> Result<AdminKeyPair, JsValue> {
+        let passphrase_bytes = pass.as_bytes();
+
+        let hc_pub_key_bytes = arg_checks::check_decode_pubkey(hc_pub_key)?;
+        arg_checks::check_passphrase(passphrase_bytes)?;
+
+        let hc_pub_key = PublicKey::from_bytes(&hc_pub_key_bytes)
+            .map_err(|e| HoloCryptoError::Custom(format!("{}", e)))?;
+        Ok(Self(
+            admin_keypair_from(hc_pub_key, email, pass)
+                .map_err(|e| HoloCryptoError::Custom(format!("keypair generation failed: {}", e)))?,
+        ))
+    }
+
+    #[wasm_bindgen]
+    /// Returns a base64 encoded signature of the payload
+    pub fn sign(&self, payload: &str) -> Result<String, JsValue> {
+        let sig_bytes = self.0.sign(payload.as_bytes()).to_bytes();
+        Ok(base64::encode_config(
+            &sig_bytes[..],
+            base64::STANDARD_NO_PAD,
+        ))
+    }
+}
+
+#[wasm_bindgen]
+pub struct WebUserKeyPair(Keypair);
+
+const WEBUSER_ARGON_ADDITIONAL_DATA: &[u8] = b"holo-crypto webuser ed25519 key v1";
+
+#[wasm_bindgen]
+impl WebUserKeyPair {
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        _email: &str,
```

Comment:
Awaiting confirmation as to exactly how these parameters should be combined/passed to the argon2 algorithm